indico/modules/events/abstracts/placeholders.py

```diff
@@ -0,0 +1,227 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2016 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from indico.util.i18n import _
+from indico.util.placeholders import Placeholder
+from indico.web.flask.util import url_for
+
+
+__all__ = ('EventTitlePlaceholder', 'EventURLPlaceholder', 'AbstractIDPlaceholder', 'AbstractTitlePlaceholder',
+           'AbstractURLPlaceholder', 'AbstractTrackPlaceholder', 'AbstractSessionPlaceholder',
+           'PrimaryAuthorsPlaceholder', 'CoAuthorsPlaceholder', 'SubmitterNamePlaceholder',
+           'SubmitterFirstNamePlaceholder', 'SubmitterLastNamePlaceholder', 'SubmitterTitlePlaceholder',
+           'TargetAbstractIDPlaceholder', 'TargetAbstractTitlePlaceholder', 'TargetSubmitterNamePlaceholder',
+           'TargetSubmitterFirstNamePlaceholder', 'TargetSubmitterLastNamePlaceholder', 'CommentsPlaceholder',
+           'ContributionTypePlaceholder', 'ContributionURLPlaceholder')
+
+
+class EventTitlePlaceholder(Placeholder):
+    name = 'event_title'
+    description = _("The title of the event")
+
+    @classmethod
+    def render(cls, abstract):
+        return abstract.event_new.title
+
+
+class EventURLPlaceholder(Placeholder):
+    name = 'event_url'
+    description = _("The URL of the event")
+
+    @classmethod
+    def render(cls, abstract):
+        return url_for('event.conferenceDisplay', abstract.event_new, _external=True)
```

Comment:
We already have `Event.url`, what about adding an `Event.external_url` property for this? I think it's the most common URL where you need `external=True` anyway